### PR TITLE
Registration page

### LINKED ISSUE #14: Registration Page
When a user visits the '/registration' path they should see a form to register.
The form should include:
- [x] Email
- [x] Password
- [x] Password Confirmation
- [x] Register Button

Once the user registers they should be logged in and redirected to the dashboard page

### PR DESCRIPTION
**Pull request by KieraAllen**
Add skeleton for registration page

- Write simple welcome page test
- Pass initial welcome page test with route, controller, and view
- Write test for registration page skeleton
- Pass initial registration page test with route, controller, and view
- TESTING CO-AUTHORING …
- Re-testing co-authored commits

Create functionality for registering a new user and saving them to the database
- Create test for registering new user 
- Add registered and logged in message
- Add post route for registering new user
- Add create action in users controller
- Add name field to test and updated password confirmation label
- Create sad path test for if user already exists
- Test for sad path message for if user already exists
- Add check for user in database
- Migrate and seed

closes #14

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
+class UsersController < ApplicationController
+  def new
+  end
+
+  def create
+# binding.pry
```
Comment (araspaas) [inline app/controllers/users_controller.rb]: We can take this out in the next PR

File: app/views/users/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+<%= content_tag :h1, "User Registration" %>
+  <%= form_tag "/register", method: :post do %>
```
Comment (smcunning) [inline app/views/users/new.html.erb:2]: Just a note as a refactor reminder for us to use a form_with later!

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+      flash[:error] = user.errors.full_messages.to_sentence
+      render :new
+    end
+  end
+
+
+  private
+
+  def user_params
+    # params.permit(:name, :email, :password)
```
Comment (araspaas) [inline app/controllers/users_controller.rb]: We can remove this comment on next PR or before we merge

File: app/models/user.rb [hunk 4]
```diff
@@ -3,5 +3,6 @@ class User < ApplicationRecord
   validates_presence_of :name,
                         :email
   validates_presence_of :password, if: :password
+  # validates :password, confirmation: true, presence: true
```
Comment (araspaas) [inline app/models/user.rb]: Was this added just as a note to see or did we forget to remove this?

File: spec/features/users/new_spec.rb [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+      fill_in :name, with: "Brian"
+      fill_in :email, with: "brian@gmail.com"
+      fill_in :password, with: "password"
+      fill_in :password_confirmation, with: "password"
+
+      click_button "Register"
+
+      user = User.last
+
+      expect(user.name).to eq("Brian")
```
Comment (araspaas) [inline spec/features/users/new_spec.rb]: I think we decided that line 31 and 33 were not needed. however it also isn't an issue.

**Review by araspaas — APPROVED** [review]
Left some comments on some comments(That was strange to type.)

**Review by smcunning — APPROVED** [review]
Everything looks great! Well done. 👍

**Pull request merged**